### PR TITLE
Adding LabelingJob to preview swagger

### PR DESCRIPTION
**Pull request by yetamsft**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (ArcturusZhang) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json:6257]: Does this need a `x-ms-enum` extension?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (ArcturusZhang) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json:6465]: Would you mind if we could change this model name to `LabelingJobProperties` to reduce the confusion? Since this is only the properties rather than the resource itself

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (ArcturusZhang) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json:6540]: Usually in a resource we put the `tags` in the root level of the model, rather than in the `properties` field. Is it possible to move this to `LabelingJobResource`?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (ArcturusZhang) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json:6552]: I am not quite sure but do we need a `x-ms-azure-resource` extension like [this](https://github.com/Azure/azure-rest-api-specs/blob/44754afdd63df701bd699c49cc74017d9dd62b75/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json#L2937) here since this is an azure resource model?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (ArcturusZhang) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json:6591]: Where is this used?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (ArcturusZhang) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json:6583]: I suppose this should be read-only?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (ArcturusZhang) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json:6592]: I suppose this should be read-only?

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]
Some minor comments, please have a look

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (yetamsft) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json:6583]: This is used in the `$skipToken` [parameter](https://github.com/yetamsft/azure-rest-api-specs/blob/44754afdd63df701bd699c49cc74017d9dd62b75/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json#L1949)of `labelingJobs_List` operation.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (yetamsft) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json:6591]: This is used in the `$skipToken` [parameter](https://github.com/yetamsft/azure-rest-api-specs/blob/44754afdd63df701bd699c49cc74017d9dd62b75/specification/machineleaarningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json#L1949)of `labelingJobs_List` operation.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (yetamsft) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json]: Removed as not used.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (yetamsft) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json]: Renamed to `LabelingJobProperties`.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (yetamsft) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json]: In the second iteration, I followed the pattern Jennny did in her PR: https://github.com/Azure/azure-rest-api-specs/pull/10421 for consistency between job models in MachineLearningServices, which has `Tags` in the inner payload other than the resource level.  Also, `Tags` and `Properties` are 2 dictionaries every resource supply. Since the `LabelingJobResource` already has a field named "Properties`, which is the "LabelingJobProperties" I feel it will be less confusing to keep these 2 dictionaries under the inner level.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (yetamsft) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json]: Yes, good catch. Added to the `LabelingJobResource`.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (yetamsft) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json]: Yes, added.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (yetamsft) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json]: Yes, added.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (ArcturusZhang) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json]: I see. Since the $skipToken parameter is the input, I suppose this continuationToken here should be an output? If so maybe this should also be readonly

**Comment by ArcturusZhang** [comment]
Hi @yetamsft since you have separated the LabelingJob entities to another swagger file, you will have to add this new swagger file to the input file list in the corresponding tag in the `readme.md` file. This will resolve the Avocado error.

And also could you please fix other CI checks?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json
Comment (yetamsft) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json]: Yes, this is marked as readonly.

**Comment by yetamsft** [comment]
> Hi @yetamsft since you have separated the LabelingJob entities to another swagger file, you will have to add this new swagger file to the input file list in the corresponding tag in the `readme.md` file. This will resolve the Avocado error.
> 
> And also could you please fix other CI checks?

Hi Arcturus, I fixed other CI failures except for the LintDiff one, which claims it met some JSON parsing error. Do you have any suggestions?

**Comment by ArcturusZhang** [comment]
Hi @jianyexi could you please have a look at the LintDiff error?

**Comment by jianyexi** [comment]
> Hi @jianyexi could you please have a look at the LintDiff error?
 It caused by the conflict of the model: $.definitions.Resource.properties.id.description' , which has defined twice in the swaggers which inside the readme tag passed to the lintDiff ,  ensure  the model has same definition in all swaggers. You can 
Find the tag in the lintDiff run log by searching 'npx autorest'

**Comment by yetamsft** [comment]
> You don't have permission to trigger SDK Automation.
> Please add yourself to _**Azure**_ group from opensource portal if you are MSFT employee,
> or please ask reviewer to add comment *** /openapibot sdkautomation ***.
> Please ask [tih@microsoft.com](mailto:tih@microsoft.com) (or NullMDR in github) for additional help.

@ArcturusZhang I've joined the _**Azure**_ github organization and still cannot trigger SDK Automation. Could you please help?

**Comment by yetamsft** [comment]
@ArcturusZhang: Thanks for triggering the build. It seems all the checks have passed. Are there any concerns we need to solve? Are we good to mark this approved to move on to the manifest change?

**Comment by ArcturusZhang** [comment]
Hi @yetamsft This PR now looks good to me, but it also needs ARM's feedback since it is introducing new APIs. I will hold the approval until they signoff to avoid accidental merging...

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/jobs.json
Comment (pilor) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/jobs.json:731]: Don't reuse top level property names.  If you need an internal "id" field name it something like "jobId".  "id" is reserved for the top level resource id of the job

**Review by pilor — APPROVED** [review]

**Review by pilor — CHANGES_REQUESTED** [review]
Accidentally approved

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/jobs.json
Comment (yetamsft) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/jobs.json]: Removed `continuationToken `.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/jobs.json
Comment (yetamsft) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/jobs.json]: Removed `Id`, the resource can be identified using the top level `Id` property.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/jobs.json
Comment (yetamsft) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/jobs.json:81]: Updated.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/jobs.json
Comment (yetamsft) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/jobs.json]: As discussed with labeling team, there is no plan to expose this operation as long running operation for now. Removed `202` result code.

**Review by pilor — APPROVED** [review]

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/jobs.json
Comment (yetamsft) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/jobs.json]: Updated to include properties in "ProxyResource".

**Comment by yetamsft** [comment]
> Hi @yetamsft This PR now looks good to me, but it also needs ARM's feedback since it is introducing new APIs. I will hold the approval until they signoff to avoid accidental merging...

**Comment by yetamsft** [comment]
@ArcturusZhang I pushed some fixes to address comments. Could you please trigger the SDK automation again?

**Comment by ArcturusZhang** [comment]
Hi @yetamsft all things should be fine for this PR, please let me know when this is ready to merge. Thanks!

**Comment by yetamsft** [comment]
> Hi @yetamsft all things should be fine for this PR, please let me know when this is ready to merge. Thanks!

Thanks, will let you know.

**Comment by yetamsft** [comment]
> Hi @yetamsft all things should be fine for this PR, please let me know when this is ready to merge. Thanks!

Hi ArcturusZhang, this PR is ready to merge. Please help merge and close this. Thanks.

**Comment by ArcturusZhang** [comment]
@yetamsft could you please resolve the merge conflict? I will merge this once the conflicts are resolved and all CIs are green

**Comment by yetamsft** [comment]
> @yetamsft could you please resolve the merge conflict? I will merge this once the conflicts are resolved and all CIs are green

Apologize for the delay. I just resolved the conflict. Please help merge this. Thanks -Ye

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**
